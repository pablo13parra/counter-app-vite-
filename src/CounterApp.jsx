import { useState } from 'react';
import PropTypes from 'prop-types';


// const csname = (a, b) => (a + b)

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        // setCounter(counter + 1);
        setCounter((c) => c + 1);
    }

    const handleSubstract = () => {
        // setCounter(counter - 1);
        setCounter((c) => c - 1);
    }

    const handleReset = () => {
        setCounter(value);
        // setCounter((c) =>  c = value );
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button aria-label="btn-inc" onClick={handleAdd}> +1 </button>
            <button aria-label="btn-dec" onClick={handleSubstract}> -1 </button>
            <button aria-label="btn-reset" onClick={handleReset}> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number,
}