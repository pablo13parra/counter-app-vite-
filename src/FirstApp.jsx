import PropTypes from 'prop-types';

// const csname = (a, b) => (a + b)

export const FirstApp = ({ title, subtitle, name }) => {
    // if (!title) { throw new Error('NO EXISTE EL TITULO, SOS DE MADERA') }

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    name: PropTypes.string
}

FirstApp.defaultProps = {
    subtitle: 'no hay',
    name: 'xd'
}