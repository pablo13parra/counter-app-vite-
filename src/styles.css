body, html {
    background-color: #000;
    color: #FFF;
    text-align: center;
}

button {
    padding: 15px;
    font-size: 3em;
    margin-right: 5px;
}