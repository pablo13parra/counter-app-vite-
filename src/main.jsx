import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorld';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={20}/>
        {/* <FirstApp title="Soy Vegeta"/> */}
    </React.StrictMode>
)