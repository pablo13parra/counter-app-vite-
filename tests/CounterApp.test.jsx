import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from "../src/CounterApp";

describe('CounterApp', () => {
    const initialValue = 10;

    test('should match with snapshot', () => {
        const { container } = render(<CounterApp value={initialValue} />);
        expect(container).toMatchSnapshot();
    });

    test('should show the initial value = 100 ', () => {
        render(<CounterApp value={100} />);
        expect(screen.getByText('100')).toBeTruthy();
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100');
    });

    test('should increase with +1 button', () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('11')).toBeTruthy();
    });
    test('should decrease with -1 button', () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('9')).toBeTruthy();
    });
    test('should reset button return to initial value', () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }))
        // fireEvent.click(screen.getByText('Reset'));
        expect(screen.getByText(initialValue)).toBeTruthy();
    });
});