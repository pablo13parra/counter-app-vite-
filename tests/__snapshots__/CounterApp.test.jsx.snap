// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should match with snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    aria-label="btn-inc"
  >
     +1 
  </button>
  <button
    aria-label="btn-dec"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
